apollo-compiler/src/test/validation/schema/reserved-enum-value-names.graphql

```diff
@@ -9,3 +9,17 @@ enum Enum {
 enum Enum2 {
   type @targetName(name: "type")
 }
+
+enum Enum3 {
+  foo @targetName(name: "type")
+}
+
+enum Enum4 {
+  foo
+  bar @targetName(name: "foo")
+}
+
+enum Enum5 {
+  foo @targetName(name: "bar")
+  bar2 @targetName(name: "bar")
+}
```

Comment:
Can you add 

```graphql
enum Enum6 {
  type @targetName(name: "newType")
}
```

I think this fails right now (but shouldn't)

Reply:
I did, doesn't seem to fail but you must be onto something? 😅

Reply:
It's all good, sorry, got a bit confused by the different branches. May I suggest a (slightly) more compact way to present things:

```kotlin
    val enumNames = mutableSetOf<String>()
    for (value in enumDefinition.enumValues) {
      val targetName = value.directives.findTargetName(schema)

      val name = targetName ?: value.name
      when {
        name.isApolloReservedEnumValueName() -> {
          val message = if (targetName == null) {
            "'$name' is a reserved enum value name, please use the @targetName directive to specify a target name"
          } else {
            "'$name' is a reserved enum value name, please use a different name"
          }
          issues.add(
              Issue.ReservedEnumValueName(
                  message = message,
                  sourceLocation = value.sourceLocation
              )
          )
        }
        name in enumNames -> {
          issues.add(
              Issue.ReservedEnumValueName(
                  message = "'${targetName}' is already defined in this enum, please use a different name",
                  sourceLocation = value.sourceLocation
              )
          )
        }
        else -> {
          // all good
          enumNames.add(name)
        }
      }
    }
```